kafka/src/test/java/com.expedia.adaptivealerting.kafka/KafkaDetectorMapperTest.java

```diff
@@ -98,25 +101,27 @@ public void testMetricDataToMappedMetricData() {
     
     /**
      * Addresses bug https://github.com/ExpediaDotCom/adaptive-alerting/issues/253
+     * See also https://stackoverflow.com/questions/51136942/how-to-handle-serializationexception-after-deserialization
      */
-    @Test(expected = StreamsException.class)
-    public void testFailsOnDeserializationException() {
-        logAndFailDriver.pipeInput(stringFactory.create(INBOUND_TOPIC, KAFKA_KEY, "invalid_input"));
-        logAndFailDriver.readOutput(OUTBOUND_TOPIC, stringDeserializer, mmdDeserializer);
+    @Test
+    public void nullOnDeserExceptionWithLogAndFailDriver() {
+        logAndFailDriver.pipeInput(stringFactory.create(INBOUND_TOPIC, KAFKA_KEY, INVALID_INPUT));
+        val record = logAndFailDriver.readOutput(OUTBOUND_TOPIC, stringDeserializer, mmdDeserializer);
```

Comment:
Looks like we are duplicating some tests from `KafkaDetectorManagerTest` can we either test the super class separately or inherit from a common test class?

Reply:
Sure, I'll look at that. Thanks.

Reply:
I didn't end up doing this. The test methods in question have several params (topics, deserializers, inputs, drivers) and by the time that all goes into an abstract test class (or even utility), the resulting code seems more complicated than what's currently here. I did streamline the implementations a bit to use utility methods for the individual test classes. My thought is that if we add those test cases to other tests then it might tip the balance in favor of extracting those. Let me know if you have any concerns. If you want me to extract them I will.